x/axelarnet/keeper/keeper.go

```diff
@@ -246,3 +269,74 @@ func (k Keeper) getFailedTransfers(ctx sdk.Context) (failedTransfers []types.IBC
 
 	return failedTransfers
 }
+
+// SetTransferCompleted sets the transfer as completed
+func (k Keeper) SetTransferCompleted(ctx sdk.Context, transferID nexus.TransferID) error {
+	t, ok := k.GetTransfer(ctx, transferID)
+	if !ok || t.Status != types.TransferPending {
+		return fmt.Errorf("transfer %s is not pending", transferID)
+	}
+
+	t.Status = types.TransferCompleted
+	k.setTransfer(ctx, t)
+
+	return nil
+}
+
+// SetTransferFailed sets the transfer as failed
+func (k Keeper) SetTransferFailed(ctx sdk.Context, transferID nexus.TransferID) error {
+	t, ok := k.GetTransfer(ctx, transferID)
+	if !ok || t.Status != types.TransferPending {
+		return fmt.Errorf("transfer %s is not pending", transferID)
+	}
+
+	t.Status = types.TransferFailed
+	k.setTransfer(ctx, t)
+
+	return nil
+}
+
+// SetTransferPending sets the transfer as pending
+func (k Keeper) SetTransferPending(ctx sdk.Context, transferID nexus.TransferID) error {
+	t, ok := k.GetTransfer(ctx, transferID)
+	if !ok || t.Status != types.TransferFailed {
+		return fmt.Errorf("transfer %s is not failed", transferID)
+	}
+
+	t.Status = types.TransferPending
+	k.setTransfer(ctx, t)
+
+	return nil
+}
+
+func getSeqIDMappingKey(portID, channelID string, seq uint64) key.Key {
+	return seqIDMappingPrefix.
```

Comment:
Might want to use `ibcPath_seq` as the key, where `ibcPath = portID/channelID`, instead to avoid a migration if we support multi hop channels in the future?

Reply:
very good point, probably require more thinking. looks like a migration when needed